### ISSUE TITLE
Reminder: Update requirements for newer packages

### ISSUE THREAD

**Issue by machinexa2** [Gallopsled/pwntools#1779]
Thanks for contributing to Pwntools!   Ideas from the community help make Pwntools an amazing tool for everybody.

pwntools==4.3.1.
  - capstone [required: >=3.0.5rc2, installed: 4.0.2]
  - intervaltree [required: >=3.0, installed: 3.1.0]
  - mako [required: >=1.0.0, installed: 1.1.4]
  - paramiko [required: >=1.15.2, installed: 2.7.2]
  - psutil [required: >=3.3.0, installed: 5.8.0]
  - pyelftools [required: >=0.2.4, installed: 0.27]
  - pygments [required: >=2.0, installed: 2.7.4]
  - pyserial [required: >=2.7, installed: 3.5]
  - requests [required: >=2.0, installed: 2.25.1]
  - ropgadget [required: >=5.3, installed: 6.5]
  - six [required: >=1.12.0, installed: 1.15.0]
  - unicorn [required: >=1.0.2rc1,<1.0.2rc4, installed: 1.0.2]
I see several packages requirement which are outdated such as capstone and pyserial. This doesn't look nice neither does it makes your package fast as outdated packages are probably less secure, consumes more memory and is slow. So requirements should keep updated pacakges as far as possible. Also some requirements are strict like unicorn required is <1.0.2rc4 which makes it very strict.

Solution:
* Using dependabot which automatically bumps version or other bots. This repo uses it: https://github.com/mayeut/pybase64/commits/master
* Upgrade code to include support of latest version of the packages. Mostly upgrade isn't required except in the case of unicorn which is very limited because of <1.0.2rc4,
* You may also just change all the requirements to latest version now and push it in new package. Later when people raise issues then you know what bugs are created due to new packages. However its not a nice way.

Close this issue when:
All the versions of requirements is greater or equal to that of my installed pacakge ( generally updated in general)

**Comment by machinexa2** [comment]
Mostly make it support with latest unicorn version

**Comment by Arusekk** [comment]
Possible duplicate of #1538.

I am quite frankly against requiring the most up-to-date package for the the art's sake. Here are the reasons:
- We have a huge dependency tree (counting everything down to leaf packages). Arguably one of the biggest dependency trees out there on pypi, so every problem that might occur (from the list below) is 100x as likely, because of our deps.
- We mostly do not lock package versions (unicorn is an exception, since they fundamentally changed some MIPS emulation during release candidate phase, which no one has time to fix now, and the fix probably requires changing our attitude, see #1538). So if you wish to install a newer version, go ahead, no worries, pip (and other package management tools) will most likely do just that for you.
- While it is possible to just tune the requirements when something bad (listed below) arises, why dare the evil?
- If someone has a project that depends on pwntools and on some other broken project, which does not support the strictly newest version yet (like we do with unicorn), they are in trouble, since thay cannot downgrade with respect to pwntools deps.
This would manifest itself easily in the opposite direction when we had a project `foooo` which would require `baaaar` and `pwntools`, with `baaaar` requiring `unicorn >= 9.9.9` and pwntools not yet supporting it. See what I mean?
- What if the newest version introduces some malfunctionality, maybe not a backdoor (which can happen, yes), but say just a simple regression we can't know yet?
- If someone is using python packages from a distribution (like Ubuntu/Debian/Kali does, like Gentoo does, like RedHat/Fedora/Mageia does, and most likely most other do), then pwntools requiring the steaming fresh packages forces the user to install another version (like with `pip install --user` or with venv). IMO this is just wasteful. And it makes packagers' work harder because of stuff like (but not limited to) the concerns in the above bullets.

I would prefer to update the 'minimum version required' to a different version iff we use a new feature that was not present before, so if you find e.g. that "pwntools does not work with intervaltree < 3" (this one is already there, because it would clearly break) this sure is a requirements bug and we will gladly fix it. We will then also probably have no objections against bumping a package requirement to the latest version for the sake of skipping the version bisection thing, but I believe that the bisected version would have more value in it. The most value would be in not mentioning the dependencies' versions at all if one would find that it works with all versions. Some projects do that and their requirements.txt looks this pure:
```
requests
pyserial
pycparser
```
The only benefit to me from having a minimum version required at all is the reduced possibility to use six/intervaltree/unicorn/... version that is outdated according to their upstream and breaking pwntools that way. Please correct me if I am wrong.

As a side note from the linguistic world, you probably see the difference between require and suggest/recommend. We recommend using the latest supported versions of the required packages.

@zachriggle WDYT? I'm just stating my opinion here, but I will respect yours if you reopen and have a point.

**Comment by zachriggle** [comment]
> I would prefer to update the 'minimum version required' to a different version iff we use a new feature that was not present before

Very much agreed, @Arusekk

@machinexa2 if you have any compelling reasons to upgrade *specific* libraries' requirements, I'd be open to hearing them -- but there's no point for (and many points against) living on the bleeding edge.  `pip install --upgrade pwntools` will generally install the latest versions anyway (if that's what you want), so the end result to the user is identical -- but pwntools will still install correctly with older versions (`pip install pwntools`, sans `--upgrade`).

For what it's worth, I almost always have the latest version of every package installed, and our official instructions in the README.md suggest that users do as well, so this Issue is effectively a no-op suggestion.

```
python3 -m pip install --upgrade pwntools
```

By that metric, most Pwntools users are already automatically getting the most up-to-date version of every dependency (except `unicorn` as explained above) -- updating our *minimum* requirements doesn't add any value for anybody, and only adds headaches.

In summary, if it ain't broke don't fix it.
